src/main/java/ca/bc/gov/chefs/etl/forms/pcd/chc/pcPatientServices/processor/PcdChcPCPSApiResponseProcessor.java

```diff
@@ -0,0 +1,148 @@
+    successResponse.setFiles(filesGenerated);
+    exchange.getIn().setBody(mapper.writeValueAsString(successResponse));
+  }
+
+  private List<PCPSChcSubmission> parsePCPSRequest(List<Root> pcpsModels)
+      throws IllegalArgumentException, IllegalAccessException, NoSuchFieldException, SecurityException {
+    List<PCPSChcSubmission> parsedChcPCPS = new ArrayList<>();
+    for (Root root : pcpsModels) {
+      PCPSChcSubmission pcpsSubmission = new PCPSChcSubmission();
+
+      pcpsSubmission.setSubmissionId(root.getForm().getSubmissionId());
+      pcpsSubmission.setCreatedAt(CSVUtil.formatDate(root.getForm().getCreatedAt()));
+      pcpsSubmission.setLateEntry(root.getLateEntry());
+      pcpsSubmission.setSubmitterFullName(root.getForm().getFullName());
+      pcpsSubmission.setSubmitterUserName(root.getForm().getUsername());
+      pcpsSubmission.setSubmitterEmail(root.getForm().getEmail());
+      pcpsSubmission.setSubmissionStatus(root.getForm().getStatus());
+      pcpsSubmission.setSubmissionVersion("" + root.getForm().getVersion());
+      pcpsSubmission.setSubmissionFormName(root.getForm().getFormName());
+      pcpsSubmission.setChcName(root.getChcName());
+      pcpsSubmission.setPcnCommunityName(root.getPcnCommunity());
+      pcpsSubmission.setHealthAuthority(root.getHealthAuthority());
+      pcpsSubmission.setFiscalYear(root.getFiscalYear());
+      pcpsSubmission.setPeriodReported(root.getPeriodReported());
+      pcpsSubmission.setReasonForExceptPeriodRep(root.getReasonForExceptionInPeriodReported());
+      pcpsSubmission.setNotes(root.getDataSubmission().getNotes());
+      List<PCPSChcSubmissionData> pcpsSubmissionDataList = new ArrayList<>();
+
+      for (Integer i = 0; i < 13; i++) {
+        PCPSChcSubmissionData pcpsSubmissionData = new PCPSChcSubmissionData();
+        Object o = root.getDataSubmission(); // The object you want to inspect
+        Class<?> c = o.getClass();
+
+        // Check if all fields are null then skip the record
+        if (c.getDeclaredField("uniquePatientsP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("uniquePatientsSinceOpeningP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("attachedToTheClinicP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("attachedNotToTheClinicP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("unattachedP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("deliveredVirtuallyPracP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("duringBusinessHoursPracP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("outsideBusinessHoursPracP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("deliveredVirtuallyNonPracP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("duringBusinessHoursNonPracP" + (i + 1)).get(o) == null
+            && c.getDeclaredField("outsideBusinessHoursNonPracP" + (i + 1)).get(o) == null) {
```

Comment:
I'd create a separate isRecordEmpty method for this. And there's room to refactor c.getDeclaredField("uniquePatientsP" + (i + 1)).get(o) a into a re-usable method. Something like:

`
private Object getPeriodicField(Object o, String name, String index) { 
    return o.getClass().getDeclaredField(name + "P" + i).get(o); 
}
`
You might find commons FieldUtils helpful somewhere as well though I checked and I don't think it adds too much value here.

Reply:
These 2 suggested refactors has been made, 2 new helper functions to determine if period is empty and to get the field name is now created.